packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -100,7 +150,38 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
     await goldenFile.writeAsBytes(imageBytes, flush: true);
   }
 
+  @override
+  Uri getTestUri(Uri key, int version) {
+    return _testingWithSkiaGold(_platform)
+      ? key
+      : _addVersion(key, version);
+  }
+
+  Uri _addPrefix(Uri golden) {
+    final String prefix = basedir.pathSegments[basedir.pathSegments.length - 2];
+    return Uri.parse(prefix + '.' + golden.toString());
+  }
+
+  Uri _addVersion(Uri key, int version) {
+    return version == null ? key : Uri.parse(
+      key
+        .toString()
+        .splitMapJoin(
+          RegExp(r'.png'),
```

Comment:
Using regexes to manipulate paths is usually not very robust. I'd suggest using the methods from package:path to take the path apart and insert the version.